pkg/auth/requests/authenticate_test.go

```diff
@@ -0,0 +1,537 @@
+		fakeProvider.name: fakeProvider,
+	}
+
+	now := time.Now()
+	userID := "u-abcdef"
+	userPrincipalID := fakeProvider.name + "_user://12345"
+
+	user := &v3.User{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userID,
+		},
+		DisplayName:  "fake-user",
+		PrincipalIDs: []string{userPrincipalID},
+	}
+
+	token := &v3.Token{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "token-v2rcx",
+			CreationTimestamp: metav1.NewTime(now),
+		},
+		Token:        "jnb9tksmnctvgbn92ngbkptblcjwg4pmfp98wqj29wk5kv85ktg59s",
+		AuthProvider: fakeProvider.name,
+		TTLMillis:    57600000,
+		UserID:       userID,
+		UserPrincipal: v3.Principal{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: userPrincipalID,
+			},
+			Me:            true,
+			PrincipalType: "user",
+			Provider:      fakeProvider.name,
+			DisplayName:   user.DisplayName,
+		},
+	}
+
+	mockIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	mockIndexer.AddIndexers(cache.Indexers{tokenKeyIndex: tokenKeyIndexer})
+	mockIndexer.Add(token)
+
+	tokenClient := &mgmtFakes.TokenInterfaceMock{
+		GetFunc: func(name string, options metav1.GetOptions) (*v3.Token, error) {
+			return token, nil
+		},
+	}
+
+	userAttribute := &v3.UserAttribute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userID,
+		},
+		GroupPrincipals: map[string]apiv3.Principals{
+			fakeProvider.name: {
+				Items: []apiv3.Principal{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: fakeProvider.name + "_group://56789",
+						},
+						MemberOf:      true,
+						LoginName:     "rancher",
+						DisplayName:   "rancher",
+						PrincipalType: "group",
+						Provider:      fakeProvider.name,
+					},
+				},
+			},
+		},
+		ExtraByProvider: map[string]map[string][]string{
+			fakeProvider.name: {
+				common.UserAttributePrincipalID: {userPrincipalID},
+				common.UserAttributeUserName:    {user.DisplayName},
+			},
+			providers.LocalProvider: {
+				common.UserAttributePrincipalID: {"local://" + userID},
+				common.UserAttributeUserName:    {"local-user"},
+			},
+		},
+	}
+	userAttributeLister := &mgmtFakes.UserAttributeListerMock{
+		GetFunc: func(namespace, name string) (*v3.UserAttribute, error) {
+			return userAttribute, nil
+		},
+	}
+
+	userLister := &mgmtFakes.UserListerMock{
+		GetFunc: func(namespace, name string) (*v3.User, error) {
+			return user, nil
+		},
+	}
+
+	userRefresher := &fakeUserRefresher{}
+
+	authenticator := tokenAuthenticator{
+		ctx:                 context.Background(),
+		tokenIndexer:        mockIndexer,
+		tokenClient:         tokenClient,
+		userAttributeLister: userAttributeLister,
+		userLister:          userLister,
+		clusterRouter:       clusterrouter.GetClusterID,
+		refreshUser:         userRefresher.refreshUser,
+	}
+
+	req := httptest.NewRequest(http.MethodGet, "/v1/namespaces", nil)
+	req.Header.Set("Authorization", "Bearer "+token.Name+":"+token.Token)
+
+	t.Run("authenticate", func(t *testing.T) {
```

Comment:
FYI tests were written in a narrative form (as opposed to a table) deliberately.